### DISCUSSION TITLE
Update failed, id xxxx doesn't exist

### DISCUSSION THREAD

**Discussion by freon27** [riderx/capacitor-updater#340] [Q&A]
I thInk this is a question rather than a bug. The updater is definitely working correctly whenever we test it, but we're also seeing a lot of these errors in Sentry.

What I'm trying to understand is the meaning of the "id xxxx doesn't exist part". Is that an indication of the cause or just a generic error message?

Thanks

**Comment by mouaadchakir** [comment]
Hi @freon27,
The error message "id xxxx doesn't exist" is usually more than just a generic error; it's a specific indication from the database or the API layer.
In the context of an updater, this typically means:
1. **Record Mismatch:** The updater is trying to modify a record with a specific ID that it expects to find in the database, but that ID is missing. This often happens if a record was deleted by another process or if there's a sync lag.
2. **Environment/Data Sync:** If you're seeing this in Sentry but not in testing, it could be that the production data has edge cases (like orphaned records) that your test environment doesn't have.
3. **Invalid ID passing:** Sometimes a null or "placeholder" ID might be accidentally passed to the update function, triggering this check.
It's essentially a "Safety Catch" to prevent the updater from running operations on non-existent data.
If this helped clarify things, please feel free to mark this as the answer!

**Reply by freon27** [reply]
Apologies if I'm wrong but this seems like it was generated with AI with no real context provided. 

It doesn't really seem to have anything to do with the question I  asked so I'm not sure what the goal is here.

**Reply by riderx** [reply]
clearly weird i agree

**Comment by riderx** [comment]
This is something it tooks us time to find and we finally fixed, basically we had a race condition that made bundle tried to be delete in two parralel flow, so sometimes they could be already deleted.
this is now resolved in latest versions

**Reply by freon27** [reply]
Thanks for the update ❤️
